tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -0,0 +1,518 @@
+	if err != nil {
+		log.Info("failed to cluster worker security group, please attach manually")
+	}
+	workerIAM, err := a.getIAMWorkerRole(infraID)
+	if err != nil {
+		log.Info("failed to cluster worker IAM role, please attach manually")
+	}
+	publicSubnetID, err := a.getPublicSubnetId(vpcID, infraID)
+	if err != nil {
+		log.Info("failed to grab a Public subnet, please attach manually")
+	}
+	sgID, err := a.findOrCreateSg(infraID, vpcID)
+	if err != nil {
+		log.Info("failed to find or create security group under VPC, please create manually")
+	} else {
+		err = a.authorizeSgIngress(sgID)
+		if err != nil {
+			log.Info("failed to attach sg: %s, please attach manually", sgID)
+		}
+	}
+	instanceID, err := a.createInstances(imageId, instanceType, sshKey, publicSubnetID, sgID, workerSG, workerIAM)
+	if err != nil {
+		return err
+	}
+	// Associate unique tags and IP address to all created instances
+	_, err = a.createInstanceNameTag(instanceID, infraID)
+	if err != nil {
+		log.Info("failed to assign name for instance: %s", instanceID)
+	}
+	ipAddr, err := a.allocatePublicIp()
+	if err != nil {
+		log.Info("failed to get a public ip from AWS, %v", err)
+	} else {
+		log.Info("waiting for instance: %s to be ready...", instanceID)
+		err = a.waitUntilInstanceStatusOk(instanceID)
+		if err != nil {
+			log.Info("error waiting for instance: %s to be ready", instanceID)
+		} else {
+			err = a.associateIPWithInstance(ipAddr, instanceID)
+			if err != nil {
+				log.Info("failed to associate an ip to instance: %s", instanceID)
+			}
+		}
+
+	}
+	err = resource.AppendInstallerInfo(&resource.InstallerInfo{
+		InstanceIDs:      []string{instanceID},
+		SecurityGroupIDs: []string{sgID},
+	}, a.resourceTrackerDir)
+	if err != nil {
+		log.Info("failed to record instance info to file at '%s', instance will not be able to be deleted, %v",
+			a.resourceTrackerDir, err)
+	}
+	log.Info("Successfully created windows instance: %s, please RDP into windows with the following:")
+	log.Info("xfreerdp /u:Administrator /v:%s  /h:1080 /w:1920 /p:'Secret2018'", ipAddr)
+	return nil
+}
+
+// DestroyWindowsVM destroys the created instance and security group on AWS specified in 'windows-node-installer.json'
+// file.
+func (a *awsProvider) DestroyWindowsVM() error {
+	// Read from `windows-node-installer.json` file.
+	instances, err := resource.ReadInstallerInfo(a.resourceTrackerDir)
+	log.Info("consuming file '%v'", a.resourceTrackerDir)
+	if err != nil {
+		return err
+	}
+	// Delete all instances from the json file.
+	for _, instanceID := range instances.InstanceIDs {
+		err = a.terminateInstance(instanceID)
+		if err != nil {
+			log.Info("failed to terminate instance: %s", instanceID)
+		}
+	}
+	// Wait for instances termination after they are initiated.
+	for _, instanceID := range instances.InstanceIDs {
+		err = a.waitUntilInstanceTerminated(instanceID)
+		if err != nil {
+			log.Info("failed to wait for instance: %s to terminate", instanceID)
```

Comment:
The message should be `time out waiting for instance to terminate` or something like that. It should be `log.Warn` or equivalent.